site/content/en/concepts/packaging/_index.md

```diff
@@ -25,4 +25,54 @@ For compatibility with other arbitrary formats, kpt supports generating
 resource configuration data from templates, configuration DSLs, and programs
 using [source functions].
 
+## Subpackages
+
+A `subpackage` is a `kpt` package which is nested within the directory tree of
+another `kpt` package.
+
+Here is an example directory structure of a `kpt` package with subpackages
+
+```sh
+foo
+├── Kptfile
+├── bar # subpackage
+│   ├── Kptfile
+│   ├── baz # subpackage
+│   │   ├── Kptfile
+│   │   └── cm.yaml
+│   └── deploy.yaml
+└── service.yaml
+```
+
+#### Use cases
+
+1. Package publishers need a way to pick and choose multiple component packages
+   which work well together, create a single single `kpt` package using them to
+   ship an out of the box application, maintain the package and abstract the
+   details from package consumers. Alternatively, package consumers may [update]
+   individual `subpackages` directly from the upstream sources.
+2. Package publishers need a way to create parameter values (ex: [setters]) to
+   be consistent across multiple `subpackages` and make it easy for package
+   consumers to provide them with a single action.
+3. (Under development) Package consumers need a way to apply a set of
+   `subpackages` in a single action to a live cluster while maintaining
+   the ability to manage them (ex: add/destroy) independently.
+
+#### Principles
+
+Here are the core principles around `subpackages` concept
+
+1. Each kpt package is an independent building block and should contain resources
+   (ex: setter definitions) in its `Kptfile`.
+2. If a package(ex: `bar`) is present in the directory tree of another package
+   (ex: `foo`), the configs of the package `bar` are out of scope for the commands
+   (only for [cfg] commands currently) run on the package `foo`.
+3. To perform same action (run a command recursively) on all the `subpackages`
```

Comment:
Just say to run a command recursively. "To perform some action" is unnecessarily vague.